src/core/server/plugins/plugin_context.ts

```diff
@@ -88,13 +89,15 @@ export function createPluginInitializerContext(
         globalConfig$: combineLatest(
           coreContext.configService.atPath<KibanaConfigType>(kibanaConfig.path),
           coreContext.configService.atPath<ElasticsearchConfigType>(elasticsearchConfig.path),
-          coreContext.configService.atPath<PathConfigType>(pathConfig.path)
+          coreContext.configService.atPath<PathConfigType>(pathConfig.path),
+          coreContext.configService.atPath<CspConfigType>(cspConfig.path)
         ).pipe(
-          map(([kibana, elasticsearch, path]) =>
+          map(([kibana, elasticsearch, path, csp]) =>
             deepFreeze({
               kibana: pick(kibana, SharedGlobalConfigKeys.kibana),
               elasticsearch: pick(elasticsearch, SharedGlobalConfigKeys.elasticsearch),
               path: pick(path, SharedGlobalConfigKeys.path),
+              csp: pick(csp, SharedGlobalConfigKeys.csp),
```

Comment:
Shouldn't we provide `directives` string as well? So other plugins can start using it without additional `createCspDirectives` call?

Reply:
This is a good point. Let me change a few things and get back to you on this.